packages/celotool/src/lib/helm_deploy.ts

```diff
@@ -1073,3 +1089,97 @@ function rollingUpdateHelmVariables() {
     )}`,
   ]
 }
+
+const celoBlockchainDir: string = path.join(os.tmpdir(), 'celo-blockchain-celotool')
+
+export async function getGenesis(celoEnv: string, reset: boolean, useMyCelo: boolean) {
+  let genesis: string = ''
+  try {
+    genesis = await getGenesisBlockFromGoogleStorage(celoEnv)
+  } catch {
+    console.debug(`Genesis file not found in GCP. Creating a new one`)
+  }
+  if (genesis === '' || reset === true) {
+    genesis = useMyCelo ? await generateMyCeloGenesis() : generateGenesisFromEnv()
+  }
+  // Upload the new genesis file to gcp
+  if (!isCelotoolHelmDryRun()) {
+    await uploadGenesisBlockToGoogleStorage(celoEnv, genesis)
+  }
+}
+
+async function generateMyCeloGenesis(): Promise<string> {
+  // Clean up the tmp dir
+  await spawnCmd('rm', ['-rf', celoBlockchainDir], { silent: true })
+  fs.mkdirSync(celoBlockchainDir)
+  const gethTag = fetchEnv(envVar.GETH_NODE_DOCKER_IMAGE_TAG)
+  const celoBlockchainVersion = gethTag.includes('.') ? `v${gethTag}` : gethTag
+  await checkoutGethRepo(celoBlockchainVersion, celoBlockchainDir)
+  await buildGethAll(celoBlockchainDir)
+
+  // Generate genesis-config from template
+  const myceloBinary = path.join(celoBlockchainDir, 'build/bin/mycelo')
+  const myceloGenesisConfigArgs = [
+    'genesis-config',
+    '--template',
+    'monorepo',
+    '--mnemonic',
+    fetchEnv(envVar.MNEMONIC),
+    '--validators',
+    fetchEnv(envVar.VALIDATORS),
+    '--dev.accounts',
+    fetchEnv(envVar.LOAD_TEST_CLIENTS),
+    '--blockperiod',
+    fetchEnv(envVar.BLOCK_TIME),
+    '--epoch',
+    fetchEnv(envVar.EPOCH),
+    '--blockgaslimit',
+    '20000000',
+  ]
+  await spawnCmdWithExitOnFailure(myceloBinary, myceloGenesisConfigArgs, {
+    silent: false,
+    cwd: celoBlockchainDir,
+  })
+
+  // TODO: Load config to customize migrations...
+
+  // Generate genesis from config
+
+  const myceloGenesisFromConfigArgs = [
+    'genesis-from-config',
+    celoBlockchainDir,
+    '--buildpath',
+    path.join(monorepoRoot, 'packages/protocol/build/contracts'),
+  ]
+  await spawnCmdWithExitOnFailure(myceloBinary, myceloGenesisFromConfigArgs, {
+    silent: false,
+    cwd: celoBlockchainDir,
+  })
+  const genesisPath = path.join(celoBlockchainDir, 'genesis.json')
+  const genesisContent = fs.readFileSync(genesisPath).toString()
+  return genesisContent
+
+  // Clean up the tmp dir as it's no longer needed
+  await spawnCmd('rm', ['-rf', celoBlockchainDir], { silent: true })
+}
+
+export async function createGethConfigConfigMap(celoEnv: string, genesisJson: string) {
```

Comment:
Given that the genesis file is too large for a config map & that this isn't used anywhere, can this be removed or do you plan to use it later?

Reply:
I do not discard exploring options to store the genesis alternatively to Google Storage in the short term, but for now I think we can remove this. Thanks for the comment.